src/store/getters.ts

```diff
@@ -36,10 +36,15 @@ export default {
   },
 
   /** Is True if there are any pending tasks or filings. */
-  hasBlocker (state: StateIF, getters): boolean {
-    let blocker = (state.hasBlockerTask || state.hasBlockerFiling || state.isCoaPending)
+  hasBlockerExceptStaffApproval (state: StateIF): boolean {
+    return (state.hasBlockerTask || state.hasBlockerFiling || state.isCoaPending)
+  },
+
+  /** Is True if there are any pending tasks or filings. */
+  hasBlocker (_state: StateIF, getters): boolean {
```

Comment:
What if you only had this getter, and in it you check if user is staff or not?

Reply:
hasBlocker method is used in many places to disable action and show html, We cannot check user is staff or not because, we need the same behaviour for staff and use except one place ```case AllowableActions.FILE_STAFF_NOTATION:```  There we need to show button

Reply:
Yes, you're right.

How about this?

allowable-actions-mixin.ts
```
      case AllowableActions.FILE_STAFF_NOTATION: {
        const blocker = (state.hasBlockerTask || state.hasBlockerFiling || state.isCoaPending)
        return (!this.isHistorical && !blocker && !!businessId && this.isRoleStaff)
      }
```

getters.ts
```
  /** Is True if there are any pending tasks or filings or the firm is not compliance. */
  hasBlocker (_state: StateIF, getters): boolean {
    let blocker = (state.hasBlockerTask || state.hasBlockerFiling || state.isCoaPending)
    // check for SP and GP not in compliance
    if (getters.isFirm && getters.isNotInCompliance) {
      blocker = true
    }
    return blocker
  }
```

Reply:
In this case, I think we have to update in both places when something need to disable

Reply:
Yes but I think it's cleaner overall.

If it was me, I would do it like this and add comments in each place referring to the other place.

Another way to think of this is in terms of compartmentalization:
- the allowable mixins checks only what it cares about
- the getter checks only what it cares about
- the mixin and getter don't depend on each other (except by code comments)